Tweaks to custom install for v0.3

samodell opened the pull request:
Addresses @RichieEscarez 's comments from PR #771

carieshmarie commented on install/Knative-custom-install.md:
```diff
@@ -66,13 +66,13 @@ service mesh. If you install any of the following options, you must install
-| **[`istio-crds.yaml`][a]** | Creates CRDs before installing Istio.                                  |
-| **[`istio.yaml`][b]**      | Install Istio with service mesh enabled (automatic sidecar injection). |
-| [`istio-lean.yaml`][c]     | Install Istio and disable the service mesh by default.                 |
+**Install files in _bold italics_ are the recommended standard install files suitable for most use cases.**
```
```suggestion
**The files in _bold italics_ are the recommended standard install files suitable for most use cases.**
```

I originally read "Install files" as Verb-noun, so suggest using an article to make it clearer.

carieshmarie commented on install/Knative-custom-install.md:
```diff
@@ -178,28 +179,28 @@ files from the Knative repositories:
-| **[`release.yaml`][5.1]**                        | Installs the following sources: [Kubernetes][6], [GitHub][6.1], [Container image][6.2], [CronJob][6.3]|
-| [`release-gcppubsub.yaml`][5.2]                  | Installs the following sources: [PubSub][6.4]                                                         |
-| [`message-dumper.yaml`][5.3]                     | Installs an Event logging service for debugging.                                                      |
+**Install files in _bold italics_ are the recommended standard install files suitable for most use cases.**
```
```suggestion
**The files in _bold italics_ are the recommended standard install files suitable for most use cases.**
```

RichieEscarez commented:
I just noticed that we removed all the dependency info. For example the way the info is not states that Eventing no longer requires Serving. Is that a new improvement for 0.3?

samodell commented on install/Knative-custom-install.md:
```diff
@@ -66,13 +66,13 @@ service mesh. If you install any of the following options, you must install
-| **[`istio-crds.yaml`][a]** | Creates CRDs before installing Istio.                                  |
-| **[`istio.yaml`][b]**      | Install Istio with service mesh enabled (automatic sidecar injection). |
-| [`istio-lean.yaml`][c]     | Install Istio and disable the service mesh by default.                 |
+**The files in _bold italics_ are the recommended standard install files suitable for most use cases.**
```
I don't know, I really like how scannable the table is this way. I do wish it was easier in markdown to show a table's legend, but I think it's hard to miss that bold line on top of the table, and as you look at each group of components, the bold italic ones jump out at you. I don't love the idea of repeating "Tip: compatible with most use cases" for each file where that's true.

How would you feel about using some other symbol (maybe †, if we can get that to show up in markdown) to denote this meaning, instead of just the formatting?

RichieEscarez replied:
Right now this line renders like a double title and the table holds 4 conventions (1.code and link for install file, 2.code bold italic and link for "common use case", 3. link to more info, 4. bold table section for component). Honestly, I think the trouble here is that our adv install guide is trying to also act as an install reference document(and help non adv users too). 

Maybe the solution is the maintain our adv. user audience focus. I think the adv user audience is going to know whether or not they want an ELK stack and what that means. To a new user, I dont think "suitable for most use cases"  (-> common use case?) is actually helpful due to the lack of detail (i immediately ask my self 'why?', what are our common use cases and are they not covered in the host specific install guides).

(temp fix) If you really want to combine new users info here too, we could clean up the table by simply adding a different indicator (†) but I really dont like the extra bold (header looking) line and would like that perhaps, moved to the end, under the table.

(long term solution) Or we could have a separate document that is focused on new users and better discuss each "use case" and what files to install.

samodell replied:
I don't think it hurts to call out what the "standard" or "common" install files are, even for advanced users. It's good for them to know what baseline they're deviating from. I went with option 1.

I feel like "a separate document focused on new users" already exists; the on-any-K8s install kind of serves that purpose. Or one of the many platform-specific guides.

samodell commented:
> I just noticed that we removed all the dependency info. For example the way the info is now states that Eventing no longer requires Serving. Is that a new improvement for 0.3?

I thought so, but I actually think that's not correct. I'll update this to indicate that Eventing still depends on Serving.

RichieEscarez commented:
> > I just noticed that we removed all the dependency info. For example the way the info is now states that Eventing no longer requires Serving. Is that a new improvement for 0.3?
> 
> I thought so, but I actually think that's not correct. I'll update this to indicate that Eventing still depends on Serving.

Yeah, we should validate all the dependency info that was in the original version to make sure they should be removed (ie. it was not just serving, another example is that you must install Serving before an observability plugin, and the new format fails to indicate that too)

samodell commented:
@RichieEscarez  My force pushed wiped the commits you made to my branch, which I was worried would happen, so I copied the changes you committed into my local copy, so they're still in the final outcome even though they've disappeared from the log here.

samodell commented:
> > > I just noticed that we removed all the dependency info. For example the way the info is now states that Eventing no longer requires Serving. Is that a new improvement for 0.3?
> > 
> > 
> > I thought so, but I actually think that's not correct. I'll update this to indicate that Eventing still depends on Serving.
> 
> Yeah, we should validate all the dependency info that was in the original version to make sure they should be removed (ie. it was not just serving, another example is that you must install Serving before an observability plugin, and the new format fails to indicate that too)

Added a "dependencies" section to the table; tried to keep it is as simple as possible.

RichieEscarez commented on install/Knative-custom-install.md:
```diff
@@ -178,33 +180,33 @@ files from the Knative repositories:
+| Knative Install Filename                         | Notes                                                                                                 | Dependencies                                                      |
+| -------------------------------------------------| ----------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------- |
+| **knative/serving**                              |                                                                                                       |                                                                   |
+| **[`serving.yaml`][1.1]_†_**                     | Installs the  Serving component.                                                                      |                                                                   |
```
```suggestion
| [`serving.yaml`][1.1]_†_                     | Installs the  Serving component.                                                                      |                                                                   |
```
remove the extra bold for all the † items in the tables and it works for me

samodell replied:
I ended up removing all the formatting from the new symbol; I don't think it needs it.

RichieEscarez commented on install/Knative-custom-install.md:
```diff
@@ -178,33 +180,33 @@ files from the Knative repositories:
+| **knative/eventing**                             |                                                                                                       |                                                                   |
+| **[`release.yaml`][4.1]_†_**                     | Installs the Eventing component. Includes the in-memory channel provisioner.                          | Serving component                                                 |
+| [`eventing.yaml`][4.2]                           | Installs the Eventing component. Does not include the in-memory channel provisioner.                  | Serving component                                                 |
+| [`in-memory-channel.yaml`][4.3]                  | Installs only the in-memory channel provisioner.                                                      | Eventing component                                                |
```
Nit: the dependency for these channel install files and any of the event sources below is both Eventing and Serving 

(just another one of those unnecessary cognitive load things where we force them to read the whole eventing row to mentally map more dependencies vs. all in one place)

samodell replied:
I added in the additional/implied dependencies.

RichieEscarez commented:
> @RichieEscarez My force pushed wiped the commits you made to my branch, which I was worried would happen, so I copied the changes you committed into my local copy, so they're still in the final outcome even though they've disappeared from the log here.

cool, no worries (thanks for mentioning it)

samodell commented:
Ready for another look, @RichieEscarez
